src/projects/projects.controller.ts

```diff
@@ -58,7 +84,13 @@ export class ProjectsController {
             );
 
             if (!project) {
-                throw new Error(`No project Found with id: ${id}!`);
+                throw new HttpException(
+                    {
+                        status: HttpStatus.NO_CONTENT,
```

Comment:
question: shouldn't this be `HttpStatus.NOT_FOUND`? Like there is no project with that id we should clearly state that this is an issue because `NO_CONTENT` reflects to 204 which is inside http's success range. Otherwise we need to adjust this inside the frontend to properly handle the case when a user trys to open an invalid project link.